### PR TITLE
chore: remove unused frontend dependencies

### PR DESCRIPTION
**Pull request by justinhynes**
[APER-943]

This PR prunes unused dependencies from the Credentials service. 

**Run JavaScript tests locally with Karma**

There is work being done on a fix to get Karma to run in CI. Until then, however, contributors are required to run these tests locally.

- [X] Make sure you are inside the devstack container
- [X] Run `make test-karma`
- [X] All tests pass

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -15,12 +15,9 @@
     "css-minimizer-webpack-plugin": "5.0.1",
     "file-loader": "6.2.0",
     "font-awesome": "4.7.0",
-    "imports-loader": "4.0.1",
```
Comment (justinhynes) [inline package.json:18]: Installed but not referenced anywhere in our webpack config.

File: package.json [hunk 2]
```diff
@@ -15,12 +15,9 @@
...
     "mini-css-extract-plugin": "2.7.6",
-    "regenerator-runtime": "0.14.0",
```
Comment (justinhynes) [inline package.json:20]: Installed but not imported or used anywhere I could see.

File: package.json [hunk 3]
```diff
@@ -15,12 +15,9 @@
...
     "sass": "1.69.5",
     "sass-loader": "13.3.2",
-    "style-loader": "3.3.3",
```
Comment (justinhynes) [inline package.json:23]: Installed but not referenced anywhere in our webpack config.

File: package.json [hunk 4]
```diff
@@ -34,14 +31,11 @@
     "@babel/preset-env": "7.23.3",
     "@edx/eslint-config": "1.3.0",
     "babel-loader": "9.1.3",
-    "edx-custom-a11y-rules": "3.0.1",
     "eslint": "8.54.0",
-    "eslint-config-airbnb": "19.0.4",
```
Comment (justinhynes) [inline package.json:39]: Installed but not referenced anywhere in our eslint config.

File: package.json [hunk 5]
```diff
@@ -34,14 +31,11 @@
...
     "eslint-plugin-import": "2.29.0",
     "eslint-plugin-jsx-a11y": "6.8.0",
     "jasmine-core": "3.99.1",
     "jasmine-jquery": "2.1.1",
-    "jquery": "3.7.1",
```
Comment (justinhynes) [inline package.json:44]: Can't find any use of jQuery.

**Review by deborahgu — APPROVED** [review]

**Pull request merged**